### ISSUE TITLE
Files missing from the "Files in this dataset" download section after save

### ISSUE THREAD

**Issue by vchendrix** [NCEAS/metacatui#569]
After creating or updating a data package with several data files the files do not show up on the metadata view page after saving and clicking "View your data package" in the  "Files in this dataset" download section.

The files do eventually show up but not always right away.  

## After saving and clicking "View your data package"
Notice that no files show up
<img width="1180" alt="screen shot 2018-04-11 at 9 31 41 am" src="https://user-images.githubusercontent.com/1558129/38630311-70dc8cc0-3d6b-11e8-9303-d02667e87f69.png">

## After refreshing the page many times
After refreshing the page many times the files slowly show up.
<img width="1178" alt="screen shot 2018-04-11 at 9 32 17 am" src="https://user-images.githubusercontent.com/1558129/38630364-8fd3d89a-3d6b-11e8-8e1e-c62732950b84.png">

**Comment by mbjones** [comment]
This is likely an indexing backlog issue. The delay is in waiting for the index queue to clear.  We encounter it frequently, and its high on our priority list to be able to scale indexing more effectively.  Can you check the indexing backlog when you see this delay to confirm that is the issue?

**Comment by vchendrix** [comment]
@mbjones I doubt there is much of a backlog as the example I posted here was from my local development box. Are there tuning parameters that could speed this up?  Also from a UX standpoint what about having the UI take this potential for a backlong into account?  Is it possible to give the user a message that the files are being indexed or some thing else that gives the user the confidence that their files weren't lost?

**Comment by mbjones** [comment]
Yes, we've had those exact same thoughts and @laurenwalker could probably point you at other tickets on this issue.  Definitely something we are working on.  There are several lines of attack: 1) speed up individual index ops so they are much faster; 2) avoid index re-work on package updates (we recheck things for safety more than needed); 3) parallelize indexing workers to scale workers with the rate of incoming data sets; 4) provide inspection APIs to know what the queue looks like to provide user feedback.  All of 1-3 are in different states of work right now.  Its going to be a bit before this gets developed and tested, and is somewhat complicated by our need to test and deploy in multiple scenarios.

**Comment by vchendrix** [comment]
@mbjones Thanks for the update!  Can you send me the issue numbers for these other tickets so that I may update the ESS-DIVE team?

**Comment by robyngit** [comment]
We should be able to use the new ResourceMapResolver introduced in v 2.34.0 to find the resource map even if it's still being indexed.
